client/compliance_client.go

```diff
@@ -585,6 +654,849 @@ func (c *complianceGRPCClient) CancelOperation(ctx context.Context, req *longrun
+	if body.GetFSfixed64() != 0 {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("fSfixed64=%v", body.GetFSfixed64()))
+	}
+	if body.GetFSint32() != 0 {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("fSint32=%v", body.GetFSint32()))
+	}
+	if body.GetFSint64() != 0 {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("fSint64=%v", body.GetFSint64()))
+	}
+	if body.GetFString() != "" {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("fString=%v", body.GetFString()))
+	}
+	if body.GetFUint32() != 0 {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("fUint32=%v", body.GetFUint32()))
+	}
+	if body.GetFUint64() != 0 {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("fUint64=%v", body.GetFUint64()))
+	}
+	if body.GetPBool() {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("pBool=%v", body.GetPBool()))
+	}
+	if body.GetPChild() != nil {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("pChild=%v", body.GetPChild()))
+	}
+	if body.GetPDouble() != 0 {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("pDouble=%v", body.GetPDouble()))
+	}
+	if body.GetPInt32() != 0 {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("pInt32=%v", body.GetPInt32()))
+	}
+	if body.GetPKingdom() != 0 {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("pKingdom=%v", body.GetPKingdom()))
+	}
+	if body.GetPString() != "" {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("pString=%v", body.GetPString()))
+	}
+	query := strings.ReplaceAll(strings.Join(queryParamStrs, "&"), " ", "+")
+
+	url := fmt.Sprintf("%s/v1beta1/repeat:bodyinfo",
+		c.endpoint,
+	)
+
+	if query != "" {
+		url += "?" + query
+	}
+
+	httpReq, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewReader(jsonReq))
+	if err != nil {
+		return nil, err
+	}
+	// Set the headers
+	for k, v := range c.xGoogMetadata {
+		httpReq.Header[k] = v
+	}
+	httpReq.Header["Content-Type"] = []string{"application/json"}
+
+	httpRsp, err := c.httpClient.Do(httpReq)
+	if err != nil {
+		return nil, err
+	}
+	defer httpRsp.Body.Close()
+
+	if httpRsp.StatusCode >= http.StatusOK {
+		return nil, fmt.Errorf(httpRsp.Status)
+	}
+
+	buf, err := ioutil.ReadAll(httpRsp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	unm := protojson.UnmarshalOptions{AllowPartial: true, DiscardUnknown: true}
+	rsp := &genprotopb.RepeatResponse{}
+
+	return rsp, unm.Unmarshal(buf, rsp)
+}
+
+// RepeatDataQuery this method echoes the ComplianceData request. This method exercises
+// sending all request fields as query parameters.
+func (c *complianceRESTClient) RepeatDataQuery(ctx context.Context, req *genprotopb.RepeatRequest, opts ...gax.CallOption) (*genprotopb.RepeatResponse, error) {
+	m := protojson.MarshalOptions{AllowPartial: true, EmitUnpopulated: true}
+	jsonReq, err := m.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+
+	queryParamStrs := []string{}
+	if req.GetInfo() != nil {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("info=%v", req.GetInfo()))
```

Comment:
Query params should be leaf fields. 

https://cloud.google.com/endpoints/docs/grpc-service-config/reference/rpc/google.api#rules-for-http-mapping:
> parameter name is the field path in the request message

(I think there's a more explicit reference, let me see whether I can dig it up)

Reply:
The internal grpc transcoding doc mentions encoding the leaf fields